### PR TITLE
Implementaion of Send Notification using Email and Sms

### PR DESCRIPTION
**Pull request by CodingIsAnArtOfMind**
Implementation of Send Notification using Email and SMS
Fetch the email from db to send a notification to a particular user by using grivience_id 
Implementing RabitMq to listen to grievance service when it updates the Message Queue.

### CODE REVIEW

File: notification-service/src/main/java/com/scaler/notificationservice/controller/NotificationController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+public class NotificationController implements NotificationOperation {
+
+    @Autowired
+    private EmailService emailService;
+
+    @Autowired
+    private SMSService smsService;
+
+
+    @Override
```
Comment (ajeetcloud) [inline notification-service/src/main/java/com/scaler/notificationservice/controller/NotificationController.java]: do you really need an api for sending notification? sending notification will be triggered by an incoming amq message

File: notification-service/src/main/java/com/scaler/notificationservice/model/NotificationMessageRequest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@JsonIgnoreType
+public class NotificationMessageRequest {
+    private  int grievance_id;
```
Comment (ajeetcloud) [inline notification-service/src/main/java/com/scaler/notificationservice/model/NotificationMessageRequest.java]: camelCase

File: notification-service/src/main/java/com/scaler/notificationservice/repository/NotificationUpdateRepository.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface NotificationUpdateRepository extends JpaRepository<Notification, Integer> {
+    Optional<Notification> findByGrievanceId(int grievanceId);
+    @Query("SELECT nu.notified_user FROM Notification nu WHERE nu.grievance_id = :grievanceId")
```
Comment (ajeetcloud) [inline notification-service/src/main/java/com/scaler/notificationservice/repository/NotificationUpdateRepository.java]: notified_user is username and not email

File: notification-service/src/main/java/com/scaler/notificationservice/controller/NotificationController.java (hunk 1 above)
Comment (CodingIsAnArtOfMind) [reply notification-service/src/main/java/com/scaler/notificationservice/controller/NotificationController.java]: Just for Testing

File: notification-service/src/main/java/com/scaler/notificationservice/model/NotificationMessageRequest.java (hunk 2 above)
Comment (CodingIsAnArtOfMind) [reply notification-service/src/main/java/com/scaler/notificationservice/model/NotificationMessageRequest.java]: Done

**Comment by CodingIsAnArtOfMind** [comment]
Facing Isuue while connecting to RabbitMq Getting time out connection error tried to change the port to 5672 from 15762 didn't resolved

**Comment by CodingIsAnArtOfMind** [comment]
Done

**Pull request closed without merging**